chandra_aca/dark_model.py

```diff
@@ -0,0 +1,237 @@
+
+    :returns: Acquisition success probability(s)
+    """
+    try:
+        date = DateTime(date).secs
+        zeros = np.atleast_1d(np.zeros_like(date + mag + t_ccd))
+        dates = date + zeros
+        t_ccds = t_ccd + zeros
+        mags = mag + zeros
+    except:
+        raise ValueError("Incompatible input shapes for 'date', 't_ccd', 'mag'")
+
+    warm_fracs = []
+    for sdate, stemp  in izip(dates, t_ccds):
+        warm_frac = get_warm_fracs(warm_threshold,
+                                   date=sdate, T_ccd=stemp)
+        warm_fracs.append(warm_frac)
+    probs = acq_success_prob(mags, warm_fracs)
+
+    if (np.array(date).ndim == 0 and np.array(t_ccd).ndim == 0
+        and np.array(mag).ndim == 0):
+           return probs[0]
+    else:
+        return probs
+
+
+
+def get_guide_success(date, t_ccd, mag):
+    """
+    Return probability of guide (bad_trak) success for given date, temperature and mag.
+
+    Any of the inputs can be scalars or arrays, with the output being the result of
+    the broadcasted dimension of the inputs.
+
+    This is based on the dark model and guide success fitting presented
+    in the State of the ACA 2013 (sot/state_of_aca/guide_acq_stats)
+
+    :param date: Date(s) (scalar or np.ndarray)
+    :param t_ccd: CD temperature(s) (degC, scalar or np.ndarray)
+    :param mag: Star magnitude(s) (scalar or np.ndarray)
+
+    :returns: Guide success probability(s)
+    """
+    raise NotImplementedError()
+
+
+
+
+def acq_success_prob(mag, warm_frac, prob_offset=0):
+    """
+    Calculate probability of acquisition success for a star with ``mag``
+    magnitude and a CCD warm fraction ``warm_frac``.  Uses the empirical relation:
+
+       P_acq_success = offset(mag) + scale(mag) * warm_frac
+
+    In ../guide_acq_success/plot_acq_success.py we find the best fit relation:
+
+      log10(scale) = 0.185 + 0.990 * (mag - 10) + -0.491 * (mag - 10)**2
+      log10(offset) = -1.489 + 0.888 * (mag - 10) + 0.280 * (mag - 10)**2
+    """
+    mag10 = mag - 10.0
+    scale = 10. ** np.polyval(acq_fit['scale'], mag10)
+    offset = 10. ** np.polyval(acq_fit['offset'], mag10)
+    # these are actually fits on failure prob, so subtract from 1.00
+    return 1.00 - (offset + scale * warm_frac - prob_offset)
+
+
+def smooth_broken_pow(pars, x):
+    """Smoothed broken power-law.  Pars are same as bpl1d (NOT + gaussian sigma):
+    1: gamma1
+    2: gamma2
+    3: x_b (break point)
+    4: x_r (normalization reference point)
+    5: ampl1
+    #   NOT 6: sigma (bins)"""
+    (gamma1, gamma2, x_b, x_r, ampl1) = pars
+    ampl2 = ampl1 * (x_b / x_r) ** (gamma2 - gamma1)
+    ok = xall > x_b
+    y = ampl1 * (xall / x_r) ** (-gamma1)
+    y[ok] = ampl2 * (xall[ok] / x_r) ** (-gamma2)
+    imin = np.searchsorted(xall, x[0] - 1e-3)
+    imax = np.searchsorted(xall, x[-1] + 1e-3)
+    return np.convolve(y, yg, mode='same')[imin:imax]
+
+
+def temp_scalefac(T_ccd):
```

Comment:
Gotcha.  Part of the reason I was wondering about keeping the dark scale piece in its own module was because dark_model here had its own copy of this temperature scaling here and I wasn't sure if you wanted to change it (or if the dark model code was more analysis code that should stay consistent with its previous outputs).  Will do.